evutil.c

```diff
@@ -2179,6 +2178,35 @@ evutil_inet_ntop(int af, const void *src, char *dst, size_t len)
 #endif
 }
 
+#ifdef _WIN32
+typedef ULONG(WINAPI *if_nametoindex_fn_t)(PCSTR);
+
+unsigned
+if_nametoindex_fn(const char *ifname)
+{
+	/* The function if_nametoindex is only available starting from
+	   Windows Vista. Checking for its existence at runtime and if
+	   it's not found, return 0.
+	*/
+	HMODULE lib = evutil_load_windows_system_library_(TEXT("iphlpapi.dll"));
+	if_nametoindex_fn_t fn;
+	unsigned result = 0;
+
+	if (!lib)
+		goto done;
+
+	if (!(fn = (if_nametoindex_fn_t)GetProcAddress(lib, "if_nametoindex")))
+		goto done;
```

Comment:
It would be better to add some log information to the exception branch.

Reply:
If it will have a log message, it should be definitely printed only once